command.js

```diff
@@ -95,39 +95,36 @@ module.exports = function(pluginOptions) {
                 );
             })
             .then(function(screenshot) {
-                var screenshotPath = getScreenshotPath(this.executionContext, screenshotId),
+                var referenceScreenshotPath = getScreenshotPath(this.executionContext, screenshotId),
+                    newScreenshotPath = getNewScreenshotPath(referenceScreenshotPath),
+                    diffScreenshotPath = getDiffPath(referenceScreenshotPath),
                     browserId = this.executionContext.browserId;
 
-                return fs.exists(screenshotPath).then(function(referenceScreenshotExists) {
+                return fs.exists(referenceScreenshotPath).then(function(referenceScreenshotExists) {
                     if(referenceScreenshotExists) {
-                        return fs.makeTmpFile().then(function(tempPath) {
-                            return saveScreenshot(screenshot, tempPath)
-                                .then(function() {
-                                    return compareScreenshots(tempPath, screenshotPath, tolerance);
-                                })
-                                .then(function(screenshotsLookSame) {
-                                    if(screenshotsLookSame) {
-                                        pluginOptions.verbose && console.log(
-                                            chalk.gray(' ... '),
-                                            chalk.green('✓'),
-                                            chalk.gray('verified ', chalk.bold(screenshotId),' screenshot in ', chalk.bold(browserId)));
-                                        return fs.remove(tempPath);
-                                    }
-                                    return saveScreenshotsDiff(tempPath, screenshotPath, getDiffPath(screenshotPath), tolerance)
-                                        .then(function() {
-                                            return fs.remove(tempPath);
-                                        })
-                                        .then(function() {
-                                            throw new Error(
-                                                'Screenshot "' +
-                                                screenshotId +
-                                                '" doesn\'t match to reference. See diff in func-test/screenshots-diff');
-                                        });
-                                });
-                        });
+                        return saveScreenshot(screenshot, newScreenshotPath)
+                            .then(function() {
+                                return compareScreenshots(newScreenshotPath, referenceScreenshotPath, tolerance);
+                            })
+                            .then(function(screenshotsLookSame) {
+                                if(screenshotsLookSame) {
```

Comment:
I think we should delete newScreenshot in this case — to make it easier to find only bad ones